framework/decode/custom_vulkan_struct_to_json.cpp

```diff
@@ -563,5 +563,58 @@ void FieldToJson(nlohmann::ordered_json& jdata, const Decoded_VkImageToMemoryCop
     }
 }
 
+void FieldToJson(nlohmann::ordered_json&                             jdata,
+                 const Decoded_VkRayTracingShaderGroupCreateInfoKHR* data,
+                 const JsonOptions&                                  options)
+{
+    if (data && data->decoded_value)
+    {
+        const VkRayTracingShaderGroupCreateInfoKHR&         decoded_value = *data->decoded_value;
+        const Decoded_VkRayTracingShaderGroupCreateInfoKHR& meta_struct   = *data;
+
+        FieldToJson(jdata["sType"], decoded_value.sType, options);
+        FieldToJson(jdata["type"], decoded_value.type, options);
+        if (decoded_value.generalShader != VK_SHADER_UNUSED_KHR)
+        {
+            FieldToJson(jdata["generalShader"], decoded_value.generalShader, options);
+        }
+        else
+        {
+            jdata["generalShader"] = "VK_SHADER_UNUSED_KHR";
```

Comment:
Does this mean that sometimes the output will be an int for a handle and sometimes will be `SHADER_UNUSED`?